docker-for-mac/osxfs.md

```diff
@@ -85,9 +85,11 @@ requests for ownership metadata will return the previously set
 values. Ownership-based permissions are only enforced at the OS X file
 system level with all accessing processes behaving as the user running
 Docker. If the user does not have permission to read extended attributes
-on an object, e.g. when that object's permissions are `0000`, ownership
-will be reported as the accessing process until the extended attribute
-is again readable.
+on an object, e.g. when that object's permissions are `0000`, `osxfs`
```

Comment:
I think this is a good simplification, thanks! I endorse Victoria's commit 415dfd7 as accurate.

As for your questions, it is not currently mentioned in the docs but this also applies to attempts to write extended attributes to persist the ownership metadata from the Linux VM. It is very likely that the reason for `chown` failure is that the UNIX permissions on the object prohibit read, write, or both which correspond to reading xattrs, writing xattrs, or both. We take advantage of a couple nice properties of the OS X FS permission model:

1. the owner of a file may change the security settings of that file regardless of the permissions
2. ACLs are included in the security settings
3. ACLs override UNIX permissions

The use case that precipitated this change was the demand for POSIX-compliance regarding the `chown`ing of read-only or mode 0 files (that are still owned by the user). By default, OS X will disallow this but now we can set a bottom priority ACL entry (or add to the first user-allow ACL entry) the ability to read and write extended attributes which then lets us treat extended attributes just like traditional UNIX ownership metadata.

The alternative, without using ACLs, is a horrible non-atomic racy dance of permission juggling as we attempt to evade the coupling of UNIX permissions and xattrs permissions.

I hope this helps elucidate the use case and issues. I'm happy to help improve these docs even more if you can think of ways they could be made clearer.

Thanks!